internal/mode/static/state/change_processor_test.go

```diff
@@ -1737,14 +1695,7 @@ var _ = Describe("ChangeProcessor", func() {
 				Expect(changed).To(BeTrue())
 			})
 
-			It("should report not changed after multiple Upserts of unrelated and unchanged resources", func() {
-				// unchanged Gateway API resources
```

Comment:
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/predicate#GenerationChangedPredicate

Because the resources are unchanged, the `generation` should remain the same and be filtered out by the controller's `GenerationChangedPredicate`.